CHANGES.rst

```diff
@@ -24,6 +24,8 @@
 
 - Add comments to ``WCSNAMEa`` header keywords. [#143]
 
+- ``makecorr`` should include STWCS-specific keywords when copying a WCS. [#150]
```

Comment:
Oh, I don't know. Should it? It's not like anyone needs a WCS on a DQ extension. It's added for convenience. Yes, for completeness we may add it. But what happens in the following sequence:

- run makecorr - it computes a WCS and updates, say the DQ extension header
- run tweakreg - it creates the extra keywords, say NMATCH
   **Does it update the ERR, DQ extensions?**
- run the database update - it updates the SCI primary WCSs.
 **It does not update the DQ, ERR, ... extensions, so they keep the WCS created by makecorr or tweakreg**
- Suppose tweakreg updated the DQ header with NMATCH. 
  The next time one runs makecorr, it updates the DQ header but does not remove the extra keywords like NMATCH.

It's very hard to keep those in sync. If. Ihad to pickj which of these cases is most harmful, it's the update from the database that doesn't update the DQ, ERR, ... headers. Also the twekreg update, if it doesn't do it.

Reply:
This is a good point (about keeping things in sync). However, I think this should work like this: tweakreg always creates a copy of the primary header if it is getting modified. So, let's say we start with an image that has `WCSNAME=IDC_1`. Then, let's assume the same WCS is primary in `DQ` and `ERR`. When `tweakreg` runs, it will create `WCSNAMEA=IDC_1` in the `SCI` extension. `DQ` and `ERR` will be already out of sync anyway. Maybe we should update `archiveWCS` (and maybe `apply_headerlet`) to try to keep things in sync in all extensions or to archive WCSes in `ERR` and `DQ` as well.

Maybe also code such as `fobj[ext].header.update(hwcs)` that applies a WCS as primary or archives a primary as an Alt should first delete existing WCS (if any) so that there will be no left-overs from previous WCS with the same key.

Reply:
We don't need alternate WCSs in any extension other than SCI. The rest of the extensions should have a copy of the primary WCS in the corresponding SCI extension.